packages/mui-material/src/Accordion/Accordion.d.ts

```diff
@@ -80,8 +78,8 @@ export type AccordionTypeMap<
        * The component used for the transition.
        * [Follow this guide](/material-ui/transitions/#transitioncomponent-prop) to learn more about the requirements for this component.
        */
-      TransitionComponent?: React.JSXElementConstructor<
-        TransitionProps & { children?: React.ReactElement<any, any> }
+      TransitionComponent?: React.ElementType<
+        TransitionProps & { children: React.ReactElement<any, any> }
```

Comment:
We are doing this in at least these components:

<img width="528" alt="image" src="https://github.com/user-attachments/assets/a1614eb8-dfcc-4a91-847f-17642480659f">

We could also fix this by:
- Make `children` required in `TransitionProps`, but that would require more changes and there are places where `children` probably is not required.
- Update transition components and make `children` optional. Again, that means more changes.

Reply:
I agree with you: when we make the change we should be consistent with it. At this point, because it's not a pressing issue (there's no users requests for it), I'm thinking we should:

- Update the original issue to include all the components in the scope
- Take it out of v6's scope

This way we avoid the unrelated breaking change of a non urgent issue.

What do you think?

Reply:
Makes sense. I'll update the issue and remove it from the v6 milestone.

Reply:
Thanks!

Reply:
Updated: https://github.com/mui/material-ui/issues/40650